ledger/acctupdates.go

```diff
@@ -2328,6 +2399,37 @@ func catchpointRoundToPath(rnd basics.Round) string {
 	return outStr
 }
 
+// This function remove a single catchpoint file from the disk. this function does not leave empty directories
+func (au *accountUpdates) removeSingleCatchpointFileFromDisk(fileToDelete string) (err error) {
+	absCatchpointFileName := filepath.Join(au.dbDirectory, fileToDelete)
+	err = os.Remove(absCatchpointFileName)
+	if err == nil || os.IsNotExist(err) {
+		// it's ok if the file doesn't exist.
+		err = nil
+	} else {
+		// we can't delete the file, abort -
+		return fmt.Errorf("unable to delete old catchpoint file '%s' : %v", absCatchpointFileName, err)
+	}
+	catchpointRootDir := filepath.Join(au.dbDirectory, catchpointDirName)
+	currentCatchpointDir := filepath.Dir(absCatchpointFileName)
+
+	for currentCatchpointDir != catchpointRootDir {
+		filesInCatchupDir, err := os.ReadDir(currentCatchpointDir)
```

Comment:
`ReadDir` is quite a slow method, since it reads all the entries. look https://stackoverflow.com/questions/30697324/how-to-check-if-directory-on-path-is-empty for alternative solution.

Reply:
Done